orderer/common/multichannel/registrar_test.go

```diff
@@ -806,6 +805,96 @@ func TestRegistrar_RemoveChannel(t *testing.T) {
+		confApp.Consortiums = nil
+		confApp.Consortium = ""
+		genesisBlockApp := encoder.New(confApp).GenesisBlockForChannel("my-non-raft-channel")
+		ledger, err := ledgerFactory.GetOrCreate("my-non-raft-channel")
+		require.NoError(t, err)
+		ledger.Append(genesisBlockApp)
+
+		registrar.CreateChain("my-non-raft-channel")
+		require.NotNil(t, registrar.GetChain("my-non-raft-channel"))
+
+		t.Run("when app channel doesn't exist", func(t *testing.T) {
+			err = registrar.RemoveChannel("some-raft-channel", false)
+			require.EqualError(t, err, "channel does not exist")
+		})
+
+		t.Run("when app channel isn't a raft channel", func(t *testing.T) {
+			err = registrar.RemoveChannel("my-non-raft-channel", false)
+			require.EqualError(t, err, "cannot remove non-raft application channel my-non-raft-channel")
+		})
+	})
+
+	t.Run("Remove channel successfully", func(t *testing.T) {
+		tmpdir, err := ioutil.TempDir("", "remove-channel")
+		require.NoError(t, err)
+		defer os.RemoveAll(tmpdir)
+
+		tlsCA, err := tlsgen.NewCA()
+		require.NoError(t, err)
+
+		confAppRaft := genesisconfig.Load(genesisconfig.SampleDevModeEtcdRaftProfile, configtest.GetDevConfigDir())
+		confAppRaft.Consortiums = nil
+		confAppRaft.Consortium = ""
+		generateCertificates(t, confAppRaft, tlsCA, tmpdir)
+		bootstrapper, err := encoder.NewBootstrapper(confAppRaft)
+		require.NoError(t, err, "cannot create bootstrapper")
+
+		ledgerFactory, _ := newLedgerAndFactory(tmpdir, "", nil)
+		consenter := &mocks.Consenter{}
+		consenter.HandleChainCalls(handleChainCluster)
+		consenter.IsChannelMemberReturnsOnCall(0, true, nil)
+		consenter.IsChannelMemberReturnsOnCall(1, false, nil)
+		// consenter.IsChannelMemberReturns(true, nil)
+		mockConsenters := map[string]consensus.Consenter{confAppRaft.Orderer.OrdererType: consenter}
+
+		config := localconfig.TopLevel{
+			General: localconfig.General{
+				BootstrapMethod: "none",
+			},
+		}
+		registrar := NewRegistrar(config, ledgerFactory, mockCrypto(), &disabled.Provider{}, cryptoProvider, nil)
+		registrar.Initialize(mockConsenters)
+
+		t.Run("as a member", func(t *testing.T) {
+			genesisBlockAppRaft := bootstrapper.GenesisBlockForChannel("my-raft-channel")
+			require.NotNil(t, genesisBlockAppRaft)
+
+			// Before join the chain, it doesn't exist
+			require.Nil(t, registrar.GetChain("my-raft-channel"))
+
+			info, err := registrar.JoinChannel("my-raft-channel", genesisBlockAppRaft, true)
+			require.NoError(t, err)
+			require.Equal(t, types.ChannelInfo{Name: "my-raft-channel", URL: "", ClusterRelation: "member", Status: "active", Height: 0x1}, info)
+			// After creating the chain, it exists
+			require.NotNil(t, registrar.GetChain("my-raft-channel"))
+
+			err = registrar.RemoveChannel("my-raft-channel", false)
+			require.NoError(t, err)
+			// After deleting the chain, it no longer exists
+			require.Nil(t, registrar.GetChain("my-raft-channel"))
```

Comment:
Check that ledger resources were removed as well.

Reply:
Done